Heatington/Controllers/CsvController.cs

```diff
@@ -59,5 +59,22 @@ public void SaveData(List<ResultHolder> data)
                 throw;
             }
         }
+
+        //TODO rewrite with generics
+        // Another overload function, it should be done with generics
+        public void SaveData(List<FormatedResultHolder> data)
+        {
+            try
+            {
+                CsvData csv = CsvData.Create(data);
+                string rawCsvData = CsvSerializer.Serialize(csv);
+                _fileController.WriteData<string>(rawCsvData);
+            }
+            catch (Exception e)
+            {
+                Utilities.DisplayException(e.Message);
+                throw;
+            }
+        }
     }
 }
```

Comment:
I am a bad coder and I also have no clue of computer science or programming in general, but before merging this, shouldn't we maybe try this out? It was already in the todo and this should be a quick change?
```suggestion
public void SaveData<T>(List<T> data)
{
    try
    {
        CsvData csv = CsvData.Create(data);
        string rawCsvData = CsvSerializer.Serialize(csv);
        _fileController.WriteData<string>(rawCsvData);
    }
    catch (Exception e)
    {
        Utilities.DisplayException(e.Message);
        throw;
    }
}
```

Reply:
PS: didn't test this.